### PR TITLE
feat: add mlflow support for SynapseML models

### PR DESCRIPTION
**Pull request by serena-ruan**
Support saving models in SynapseML as mlflow type models, thus supporting corresponding functionalities in mlflow including load_model, log_model, predict etc.

### CODE REVIEW

**Review by imatiach-msft — DISMISSED** [review]
This is awesome!!  Does this resolve the issue https://github.com/microsoft/SynapseML/issues/1275 ?
Can you add a description to the PR?  Thank you!

**Comment by serena-ruan** [comment]
> This is awesome!! Does this resolve the issue #1275 ? Can you add a description to the PR? Thank you!

Oh yes! This looks relevant! we could support log_model in the PR as well to solve that issue.

File: core/src/main/scala/com/microsoft/azure/synapse/ml/isolationforest/IsolationForest.scala [hunk 1]
```diff
@@ -28,7 +28,7 @@ class IsolationForest(override val uid: String, val that: IsolationForestSource)
 
   override def fit(data: Dataset[_]): IsolationForestModel = {
     logFit(
-      new IsolationForestModel(uid).setInnerModel(that.fit(data))
+      new IsolationForestModel(uid).setInnerModel(that.fit(data).setPredictionCol("prediction"))
```
Comment (mhamilton723) [inline core/src/main/scala/com/microsoft/azure/synapse/ml/isolationforest/IsolationForest.scala:31]: this should be set with the estimator's prediction cog

**Review by mhamilton723 — CHANGES_REQUESTED** [review]

**Review by imatiach-msft — DISMISSED** [review]

**Review by mhamilton723 — APPROVED** [review]

**Review by mhamilton723 — APPROVED** [review]

**Pull request merged**